Lib/logging/handlers.py

```diff
@@ -232,19 +232,19 @@ def __init__(self, filename, when='h', interval=1, backupCount=0,
         if self.when == 'S':
             self.interval = 1 # one second
             self.suffix = "%Y-%m-%d_%H-%M-%S"
-            self.extMatch = r"^\d{4}-\d{2}-\d{2}_\d{2}-\d{2}-\d{2}(\.\w+)?$"
+            extMatch = r"(?<!\d)\d{4}-\d{2}-\d{2}_\d{2}-\d{2}-\d{2}(?!\d)"
```

Comment:
Are the negative lookaheads just to capture the preceding and following parts around the date-time suffix? Please add a comment about why they're used. Otherwise, looks good, thanks!

Reply:
They are not capturing. They are to ensure that the suffix is not preceded or followed by digits, that decreases the chance of accident wrong matches. In "12024-03-2024-03-01", it should not match "1**2024-03-20**24-03-01", but "12024-03-**2024-03-01**". This code no longer requires that the datetime suffix is separated by dots from other parts, "foo2024-03-01bar.log" is now supported. Although, it is merely an optimization, because the following check filters out false matches.

I am not sure what kind of comment is needed here.